olog.py

```diff
@@ -16,77 +13,159 @@ def __init__(self, url, user, password):
         user : string
         password : string
         """
-        if url.endswith('/'):
-            url = url[:-1]
-        auth = requests.auth.HTTPBasicAuth(user, password)
-        self._url = url
-        self._session = requests.Session()
-        # I have requested that working certificates be properly configured on
-        # the server. This is unfortunately a necessary workaround for now....
-        self._session.verify = False
-        self._kwargs = dict(headers=headers, auth=auth)  # for every request
-
-    def list_logbooks(self):
-        url = f'{self._url}/resources/logbooks'
-        res = self._session.get(url, **self._kwargs)
-        res.raise_for_status()
-        return res.json()
+        if not url.endswith('/'):
+            url += '/'
+        self._session = AsyncClient(base_url=url, auth=(user, password))
 
-    def get_logbook(self, name):
-        # Logbooks have an integer id, but this REST endpoint expects the
-        # *name*.  It does not accept an id.
-        url = f'{self._url}/resources/logbooks/{name}'
-        res = self._session.get(url, **self._kwargs)
-        res.raise_for_status()
-        return res.json()
+    # Logbooks
+    async def aget_logbooks(self):
+        async with self._session as api:
+            res = await api.get('logbooks')
+            res.raise_for_status()
```

Comment:
Here and everywhere below, it is best to exit from the `with` block as early as you can in order to release system resources.